### PR TITLE
[New] Add post-robot w/ git auto-update

### LINKED ISSUE #9977: [Request] Add post-robot
post-robot
https://github.com/krakenjs/post-robot
https://www.npmjs.com/package/post-robot

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/40096776-request-add-post-robot?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F32893&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by dakshshah96**
Pull request for issue: #9977 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/krakenjs/post-robot
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/post-robot
 * License and its reference: Apache-2.0
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 18
   - Count of stars: 88 
   - Count of forks: 14
 * NPM download stats (optional):
   - Downloads in the last day: 1,825
   - Downloads in the last week: 4,843
   - Downloads in the last month: 12,864

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by dakshshah96** [comment]
@sufuf3 Done 👍

**Comment by sufuf3** [comment]
@dakshshah96 Could you revise the first commit message from `npm` to `git`?
The first commit message should modify as the following because we use git auto-update:
```
Add post-robot@7.0.2 w/ git auto-update

close #9977, close #10183, cc #11170, cc @bluepnume
```

Except for this part, I think this PR is fine.

Thank you.

**Comment by sufuf3** [comment]
Oh~ it a little weird about `index.min.js`.
I run the `web-minify-helper/minify.sh` but the file content which I generated is different from yours.

**Comment by dakshshah96** [comment]
@sufuf3 Modified the commit message. Even I ran `web-minify-helper` to minify all the JS files.

File: ajax/libs/post-robot/0.0.1/index.min.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<html>
+<head><title>301 Moved Permanently</title></head>
+<body bgcolor="white">
+<center><h1>301 Moved Permanently</h1></center>
+<hr><center>nginx/1.4.6 (Ubuntu)</center>
+</body>
+</html>
```
Comment (sufuf3) [inline ajax/libs/post-robot/0.0.1/index.min.js]: But I don't think this is the `index.min.js` real content.
This looks like the HTML file.
How about remove the `index.min.js` and run again?
Thank you.

**Comment by dakshshah96** [comment]
@sufuf3 Done 👍

File: ajax/libs/post-robot/package.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "postmessage",
+    "krakenjs",
+    "kraken"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/krakenjs/post-robot.git"
+  },
+  "license": "Apache-2.0",
+  "author": "Daniel Brain <@bluepnume>",
```
Comment (extend1994) [inline ajax/libs/post-robot/package.json]: Please modify this as `Daniel Brain <dbrain@paypal.com>`
Ref: https://github.com/krakenjs/post-robot/blob/master/bower.json#L6

**Comment by dakshshah96** [comment]
@sufuf3 Done 👍

@extend1994 I've changed the `author` field. Can you please confirm if v0.0.1~v1.0.8 are compatible with a browser or not? cc @sufuf3

**Comment by sufuf3** [comment]
@dakshshah96 Would you modify `"author": "Daniel Brain <@bluepnume>"` to `"author": "Daniel Brain <dbrain@paypal.com>"` as @extend1994 said https://github.com/cdnjs/cdnjs/pull/11449#discussion_r125186020 ?

Thank you.

**Comment by sufuf3** [comment]
@cdnjs/intern3 @cdnjs/library-reviewer Please help review this PR, thanks.
It stuck a little longer. 😢

**Comment by sufuf3** [comment]
@sashberd I know it's good to do this.
But actually, it stuck a little longer, If reviewer require to modify the PR and it be stagnated by contributor, I think it's great to do this.
But if contributor do well on this PR and it's just bogged down by reviewers, I prefer not to bother contributor if auto-update config is fine and other files also good.

Also, after this PR be merged, all versions  of the lib will be added.

What do you think?

Many thanks.

**Comment by maruilian11** [comment]
@sufuf3 Have we checked the previous version compatibility problem? Looks like it can.

**Comment by maruilian11** [comment]
Although it mentioned `MIT` in bower.json. We can see this commit modify the license to `Apache-2.0`.
https://github.com/krakenjs/post-robot/commit/71d7df42aabd03c11f227304361384453b264da6

**Comment by bluepnume** [comment]
I've fixed the license in bower.json to `Apache 2.0`

**Comment by sufuf3** [comment]
@dakshshah96 thanks for contribution.
Would you mind working on this and helping improve this PR?
If you have any question, please let us know.

Many thanks.

**Comment by sashberd** [comment]
@PeterDaveHello I helped with this one. FYR

**Comment by PeterDaveHello** [comment]
@sashberd I didn't see it

**Comment by sashberd** [comment]
@PeterDaveHello Hmmm... May be something with Ethernet was wrong. Anyway, now updated. FYR

**Pull request closed without merging**